### PR TITLE
Wheels: ADIOS2 -> v2.9.0, blosc2

### PR DESCRIPTION
**Pull request by franzpoeschel**
ADIOS2 v2.9 replaces c-blosc with c-blosc2 as a dependency

TODO:

- [x] Await final release of ADIOS2 v2.9
- [x] Also update HDF5 to 1.14+ for H5Vol layer updates in ADIOS2 v2.9?
  - [x] or disable HDF5 in ADIOS2 for now
- [x] Windows
- [x] Python 3.8+ (3.7 went EOL last month)

### CODE REVIEW

File: library_builders.sh [hunk 1]
```diff
@@ -203,6 +211,43 @@ function build_blosc {
...
+    if [ -e blosc-stamp2 ]; then return; fi
+
+    curl -sLo c-blosc2-v2.7.1.tar.gz \
+        https://github.com/Blosc/c-blosc2/archive/refs/tags/v2.7.1.tar.gz
+    file c-blosc2*.tar.gz
+    tar -xzf c-blosc2*.tar.gz
+    rm c-blosc2*.tar.gz
+
+    # @todo: proper patch for this
+    sed -E -i.bak 's|if\(\$ENV\{CMAKE_OSX_ARCHITECTURES\} STREQUAL "arm64"\)|if("$ENV{CMAKE_OSX_ARCHITECTURES}" STREQUAL "arm64")|' c-blosc2-*/CMakeLists.txt
```
Comment (ax3l) [inline library_builders.sh]: @franzpoeschel I do not see why that env var needs to be in quotes. Can you elaborate? :)
https://github.com/Blosc/c-blosc2/blob/v2.7.1/CMakeLists.txt#L243

File: library_builders.sh [hunk 2]
```diff
@@ -203,6 +211,43 @@ function build_blosc {
     touch blosc-stamp
 }
 
+function build_blosc2 {
+    if [ -e blosc-stamp2 ]; then return; fi
+
+    curl -sLo c-blosc2-v2.7.1.tar.gz \
```
Comment (ax3l) [inline library_builders.sh]: @franzpoeschel Is there a reason you selected 2.7.1 and not the latest 2.9.3 for c-blosc2?

Comment (ax3l) [reply library_builders.sh:217]: Oh, I see:
```
  -- Using ZSTD internal sources.
      "STREQUAL" "arm64"
```

wild!

File: library_builders.sh (hunk 1 above)
Comment (ax3l) [reply library_builders.sh]: Oh, I see:
```
  -- Using ZSTD internal sources.
      "STREQUAL" "arm64"
```

wild!

Comment (ax3l) [reply library_builders.sh]: X-ref: https://github.com/Blosc/c-blosc2/pull/525

**Comment by ax3l** [comment]
Windows link error in ADIOS2:
```
2023-07-01T02:44:28.5724342Z LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
2023-07-01T02:44:28.6284514Z libblosc2.lib(blosc2.obj) : error LNK2019: unresolved external symbol compress2 referenced in function blosc_c [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
2023-07-01T02:44:28.6285930Z libblosc2.lib(blosc2.obj) : error LNK2019: unresolved external symbol uncompress referenced in function blosc_d [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
2023-07-01T02:44:28.6575523Z D:\a\openPMD-api\openPMD-api\src\build-adios2\bin\Debug\adios2_reorganize.exe : fatal error LNK1120: 2 unresolved externals [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\adios_reorganize.vcxproj]
2023-07-01T02:44:29.4429936Z   Engine.cpp
2023-07-01T02:44:29.4431298Z LINK : warning LNK4098: defaultlib 'MSVCRT' conflicts with use of other libs; use /NODEFAULTLIB:library [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
2023-07-01T02:44:29.5698764Z libblosc2.lib(blosc2.obj) : error LNK2019: unresolved external symbol compress2 referenced in function blosc_c [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
2023-07-01T02:44:29.5704186Z libblosc2.lib(blosc2.obj) : error LNK2019: unresolved external symbol uncompress referenced in function blosc_d [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
2023-07-01T02:44:29.5725378Z D:\a\openPMD-api\openPMD-api\src\build-adios2\bin\Debug\bpls.exe : fatal error LNK1120: 2 unresolved externals [D:\a\openPMD-api\openPMD-api\src\build-adios2\source\utils\bpls.vcxproj]
```
might be from ZLIB in c-blosc2?
```
...
2023-07-01T02:33:57.9231340Z -- Could NOT find ZLIB_NG (missing: ZLIB_NG_LIBRARY ZLIB_NG_INCLUDE_DIR) 
2023-07-01T02:33:57.9741468Z -- Found ZLIB: C:/Program Files (x86)/zlib/lib/zlibstatic.lib (found version "1.2.13")  
...
```
https://github.com/Blosc/c-blosc2/pull/529

File: library_builders.sh [hunk 3]
```diff
@@ -287,13 +342,12 @@ function build_hdf5 {
         HOST_ARG="--host=aarch64-apple-darwin"
 
         curl -sLo osx_cross_configure.patch \
-            https://raw.githubusercontent.com/h5py/h5py/fcaca1d1b81d25c0d83b11d5bdf497469b5980e9/ci/osx_cross_configure.patch
+            https://raw.githubusercontent.com/conda-forge/hdf5-feedstock/ca7b3b40d58cb949638ead75c9527b3fd1112a3f/recipe/patches/osx_cross_configure.patch
```
Comment (ax3l) [inline library_builders.sh]: Ok, the only problem left here is that these two patches for H5detect to use the host native version do not seem to work yet (macOS arm64).

Probably needs to wait until after my vacation.

File: library_builders.sh (hunk 2 above)
Comment (franzpoeschel) [reply library_builders.sh]: I recall picking the most recent version that I saw. So, no reason probably, no

File: library_builders.sh (hunk 3 above)
Comment (ax3l) [reply library_builders.sh]: We will stay with HDF5 1.12.2 for now on Unix to move on.

**Review by ax3l — APPROVED** [review]
Whuiii, ok we got it - awesome :)

File: library_builders.bat [hunk 4]
```diff
@@ -193,7 +242,8 @@ call :install_buildessentials
 call :build_zlib
 :: build_bzip2
 :: build_szip
-call :build_blosc
 call :build_zfp
+call :build_blosc
```
Comment (franzpoeschel) [inline library_builders.bat]:
```suggestion
```

I don't see ADIOS1 being built in this file, so Blosc1 should probably be removed, too?

File: library_builders.sh [hunk 5]
```diff
@@ -339,11 +371,16 @@ fi
 
 install_buildessentials
 build_zlib
-build_blosc
```
Comment (franzpoeschel) [inline library_builders.sh:342]: Here it's probably the opposite. Since this still builds ADIOS1, should be put Blosc1 back in?

**Review by franzpoeschel — COMMENTED** [review]
Great progress, thank you for this, Axel! I think there is still a little inconsistency in where Blosc1 is built by this, otherwise this looks good.

**Comment by franzpoeschel** [comment]
The failing Travis jobs look like they're timing out? 
Should we worry about the failing Readthedocs run?

File: library_builders.bat (hunk 4 above)
Comment (ax3l) [reply library_builders.bat]: Correct, good catch! :)

File: library_builders.sh (hunk 5 above)
Comment (ax3l) [reply library_builders.sh:342]: Discussed in person:
We always skipped ADIOS1 on mac x-compile. Now we skip it also for Linux with ppc64le.

Forgot to remove Blosc1 on Windows (never had ADIOS1 support).

**Comment by ax3l** [comment]
> The failing Travis jobs look like they're timing out?

Looks like sporadic fails, which are unfortunately normal and can be restarted: serverl builds for PPC64le musllinux and manylinux pass.

>  Should we worry about the failing Readthedocs run?

I did not find yet how we can exclude the `wheels` branch from RTD CI builds :sweat:

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
Ah, also time to remove Python 3.7: https://endoflife.date/python

**Review by ax3l — APPROVED** [review]

**Comment by ax3l** [comment]
For the 0.15.2 release, we will need to patch in https://github.com/ornladios/ADIOS2/pull/3769 here

**Pull request merged**